lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp

```diff
@@ -3,14 +3,48 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "ttmlir/Dialect/TTIR/Analysis/LegalGridAnalysis.h"
+#include "ttmlir/Dialect/TTIR/IR/TTIROps.h"
 
 namespace mlir::tt::ttir {
 
+bool mock_is_output_tensor_legal_for_op(Operation *op, LayoutAttr layout) {
+  // Placeholder, needs to be replaced with a call the the TTNN op interface.
+  return true;
+}
+
+bool tensor_shape_compatible_with_shard(Operation *op, LayoutAttr layout) {
+  // These constraints are implemented seperatelly in every TTNN op.
+  // Almost nothing seems to be shared between EVERY op, so is hard to have any
+  // logic here without the risk of discarding a valid configuraiton or modeling
+  // the constraint for each op. This logic may be offloaded to the TTNN op
+  // interface.
+
+  // For now we will check if the tilised tensor dims are divisible by the grid
+  // dims. This will definitly discard possible valid configurations, but is a
+  // start.
+  RankedTensorType tensorType =
+      mlir::cast<RankedTensorType>(op->getResult(0).getType());
+  llvm::ArrayRef<int64_t> tensorShape = tensorType.getShape();
+
+  int64_t MTiles = 1;
+  if (tensorType.getRank() >= 2) {
+    MTiles = (tensorShape.rbegin()[1] + 31) / 32;
+  }
+
+  int64_t KTIles = (tensorShape.back() + 31) / 32;
+
+  int64_t gridR = layout.getGrid().getShape()[0];
+  int64_t gridC = layout.getGrid().getShape()[1];
+
+  return (MTiles % gridR == 0) && (KTIles % gridC == 0);
+}
+
 bool LegalGridAnalysis::applyOverrides() {
   // Lookup grid size overrides based on location information for current
   // operation.
   //
 
+  // TODO(odjuricic): Need to override all params, not just grid size.
```

Comment:
Is this for followup change? File an issue under Optmizier if so.

Reply:
Yes, there is an issue filed.